src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.swing.sources;
+
+import org.junit.Test;
+import rx.Subscription;
+import rx.functions.Action0;
+import rx.observers.TestSubscriber;
+
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+
+import static junit.framework.Assert.assertEquals;
+
+public class ListSelectionEventSourceTest {
+
+    @Test
+    public void jtableRowSelection_observingSelectionEvents() throws Throwable {
+        SwingTestHelper.create().runInEventDispatchThread(new Action0() {
+
+            @Override
+            public void call() {
+                TestSubscriber<ListSelectionEvent> testSubscriber = TestSubscriber.create();
+
+                JTable table = createJTable();
+                ListSelectionEventSource
+                        .fromListSelectionEventsOf(table.getSelectionModel())
+                        .subscribe(testSubscriber);
+
+                testSubscriber.assertNoErrors();
+                testSubscriber.assertNoValues();
+
+                table.getSelectionModel().setSelectionInterval(0, 0);
+
+                testSubscriber.assertNoErrors();
+                testSubscriber.assertValueCount(1);
+
+                assertListSelectionEventEquals(
+                        new ListSelectionEvent(
+                                table.getSelectionModel(),
+                                0 /* start of region with selection changes */,
```

Comment:
Thanks for the comments here. We should add more descriptions like this to the rest of our tests.